service/src/main/java/uk/nhs/adaptors/gp2gp/ehr/mapper/diagnosticreport/ObservationMapper.java

```diff
@@ -103,21 +106,41 @@ public String mapObservationToCompoundStatement(Observation observationAssociate
     }
 
     private String mapObservationToNarrativeStatement(IdMapper idMapper, Observation observation) {
-        var narrativeStatementTemplateParameters = NarrativeStatementTemplateParameters.builder()
-            .narrativeStatementId(idMapper.getOrNew(ResourceType.Observation, observation.getIdElement()))
-            .commentType(prepareCommentType(observation).getCode())
-            .issuedDate(DateFormatUtil.toHl7Format(observation.getIssued().toInstant()))
-            .comment(observation.getComment())
-            .availabilityTimeElement(StatementTimeMappingUtils.prepareAvailabilityTimeForObservation(observation));
+        StringBuilder narrativeStatementsBlock = new StringBuilder();
 
-        if (observation.hasPerformer()) {
-            final String participantReference = idMapper.get(observation.getPerformerFirstRep());
-            final String participantBlock = participantMapper
-                .mapToParticipant(participantReference, ParticipantType.PERFORMER);
-            narrativeStatementTemplateParameters.participant(participantBlock);
+        if (observation.hasComment()) {
+            CommentType commentType = prepareCommentType(observation);
+
+            narrativeStatementsBlock.append(
+                prepareNarrativeStatement(observation, observation.getComment(), commentType.getCode(), idMapper)
+            );
         }
 
-        return TemplateUtils.fillTemplate(NARRATIVE_STATEMENT_TEMPLATE, narrativeStatementTemplateParameters.build());
+        CodeableConceptMappingUtils.extractTextOrCoding(observation.getDataAbsentReason()).ifPresent(dataAbsentReason ->
+            narrativeStatementsBlock.append(
+                prepareNarrativeStatement(observation, dataAbsentReason, CommentType.LABORATORY_RESULT_DETAIL.getCode(), idMapper)
+            )
+        );
+
+        CodeableConceptMappingUtils.extractTextOrCoding(observation.getInterpretation()).ifPresent(interpretation ->
+            narrativeStatementsBlock.append(
+                prepareNarrativeStatement(observation, interpretation, CommentType.LABORATORY_RESULT_DETAIL.getCode(), idMapper)
+            )
+        );
+
+        CodeableConceptMappingUtils.extractTextOrCoding(observation.getBodySite()).ifPresent(bodySite ->
+            narrativeStatementsBlock.append(
```

Comment:
bodySite should start with Site prefix,  "Site: { bodySite.text or bodySite.coding[0].display }" and also time is required, please check excel line 158 column B